niworkflows/interfaces/mni.py

```diff
@@ -231,18 +354,115 @@ def _validate_results(self):
 
 
 def mask(in_file, mask_file, new_name):
+    """
+    Apply a binary mask to an image.
+
+    Parameters
+    ----------
+    in_file : str
+        Path to a NIfTI file to mask
+    mask_file : str
+        Path to a binary mask
+    new_name : str
+        Path/filename for the masked output image.
+
+    Returns
+    -------
+    str
+        Absolute path of the masked output image.
+
+    Notes
+    -----
+    in_file and mask_file must be in the same
+    image space and have the same dimensions.
+    """
     import nibabel as nb
     import os
-
+    # Load the input image
     in_nii = nb.load(in_file)
+    # Load the mask image
     mask_nii = nb.load(mask_file)
+    # Set all non-mask voxels in the input file to zero.
     data = in_nii.get_data()
     data[mask_nii.get_data() == 0] = 0
+    # Save the new masked image.
     new_nii = nb.Nifti1Image(data, in_nii.affine, in_nii.header)
     new_nii.to_filename(new_name)
     return os.path.abspath(new_name)
 
 
+def create_cfm(in_file, lesion_mask=None, global_mask=True, out_path=None):
+    """
+    Create a mask to constrain registration.
+
+    Parameters
+    ----------
+    in_file : str
+        Path to an existing image (usually a mask).
+        If global_mask = True, this is used as a size/dimension reference.
+    out_path : str
+        Path/filename for the new cost function mask.
+    lesion_mask : str, optional
+        Path to an existing binary lesion mask.
+    global_mask : bool
+        Create a whole-image mask (True) or limit to reference mask (False)
+
+    Returns
+    -------
+    str
+        Absolute path of the new cost function mask.
+
+    Notes
+    -----
+    in_file and lesion_mask must be in the same
+    image space and have the same dimensions
+    """
+    import os
+    import numpy as np
+    import nibabel as nb
+    from nipype.utils.filemanip import fname_presuffix
+
+    if out_path is None:
+        out_path = fname_presuffix(in_file, suffix='_cfm', newpath=os.getcwd())
+    else:
+        out_path = os.path.abspath(out_path)
+
+    if not global_mask and not lesion_mask:
+        NIWORKFLOWS_LOG.warning(
+            'No lesion mask was provided and global_mask not requested, '
+            'therefore the original mask will not be modified.')
+
+    # Load the input image
+    in_img = nb.load(in_file)
+
+    # If we want a global mask, create one based on the input image.
+    if global_mask:
```

Comment:
```Python
data = np.ones(in_img.shape, dtype=np.uint8)  # Set a default whole-image mask

if not global_mask:  # Rewrite mask if not global
    data = np.asanyarray(in_img.get_data() > 0, dtype=np.uint8)
```

Reply:
I guess I don't see a value in allocating an array we'll have to throw out. I guess it's only 3D, so that's NBD, but how about we just use a ternary operator?

```Python
data = np.ones(in_img.shape, dtype=np.uint8) if global_mask else in_img.get_data()
```